File: 10_June_2nd/Gyeongwook/longestCommonPrefix.py
class Solution:
    # @param {string[]} strs
    # @return {string}
    def longestCommonPrefix(self, strs):
        if len(strs) == 0:
            return ""
            
        common_prefix = strs[0]
        
        for each_str in strs:
            for idx in range(len(common_prefix)):
                if idx >= len(each_str) or each_str[idx] != common_prefix[idx]:
                    common_prefix = common_prefix[0:idx]
                    break;

        return common_prefix

        

File: 10_June_2nd/JongYoon/LongestCommonPrefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0)
            return "";
        else if (strs.length == 1)
            return strs[0];
        
        int prefixNum = strs[0].length() - 1;
        int prefix = -1;
        
        for( int i = 1 ; i < strs.length ; i++)
        {
            prefix = -1;
            
            for(int j = 0 ; j <= prefixNum && j < strs[i].length() && j < strs[i-1].length() ; j++ )
            {
                if(strs[i-1].charAt(j) == strs[i].charAt(j)) prefix = j;
                else break;
            }
            
            if ( prefix == -1 )
                return "";
                
            prefixNum = prefix;
        }
       
        return strs[0].substring(0,prefixNum+1);
    }
}